WordPress/gutenberg
# Patterns: Allow inserting of unsynced patterns from quick inserter

## What?
Allows adding of user added unsynced patterns from the quick inserter

## Why?
Currently it is confusing that a user can add synced patterns and insert them from the quick inserter, but not unsynced patterns

## How?
Adds the syncStatus to the inserter reusable blocks so unsynced patterns can be identified and handled differently

There is an [additional PR](https://github.com/WordPress/gutenberg/pull/52911) which merges the Synced patterns  inserter tab into the My patterns section of the Patterns tab.

## Testing Instructions

- Add some synced and unsynced patterns in the post editor
- Make sure all of them can be accessed via the quick inserter by name
- Make sure that only synced patterns appear in the Sync patterns inserter tab, and unsynced in the patterns tab still

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/04a92e8f-f77f-499a-a418-70428e8f9a99
